eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/MockableArmClientProvider.cs

```diff
@@ -56,5 +64,70 @@ private MethodProvider BuildGetResourceIdMethodForResource(ResourceClientProvide
 
             return new MethodProvider(signature, body, this);
         }
+
+        private IList<MethodProvider> BuildMethodsForExtensionResource(ResourceClientProvider resource)
+        {
+            var result = new List<MethodProvider>();
+
+            var scopeParameter = new ParameterProvider("scope", $"The scope of the resource collection to get.", typeof(ResourceIdentifier));
+            var signature = new MethodSignature(
+                $"Get{resource.Name}s",
```

Comment:
I think the "get collection" method is using the resource name without resource suffix
therefore it should be `Get{resource.ResourceName.Pluralize()}`.

Reply:
Correction:
the method is not in the mockableresourceprovider.
Each resourceClientProvider has a `FactoryMethodSignature` which just returns this signature to you.
We should update that method in resource (which it should know it is an extension resource), and call it here.
The implementation is just passing through all the parameters.